### PR TITLE
Krzeelzb assignment 2

### PR DESCRIPTION
**Pull request by krzeelzb**
Krzeelzb assignment 2

### CODE REVIEW

File: krzeelzb/assignment-2/src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        System.out.println(nodeInATree.getAncestor().getValue());
+
+        System.out.println("Find ancestor");
+        List<Node> ancestors=FindAncestors.findAncestors(tree,nodeInATree);
+        ancestors.forEach(s-> System.out.println(s.getValue()));
+
+        System.out.println("Common ancestors");
+        Node sample=CommonAncestor.getCommonAncestor(tree,nodeInATree,nodeInATreeTwo);
+        System.out.println(sample.getValue());
+
```
Comment (imorlowska) [inline krzeelzb/assignment-2/src/main/java/Main.java]: Very nice! How about adding a few more test cases? :)

File: krzeelzb/assignment-2/src/main/java/Node.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        this.value = value;
+        this.ancestor = ancestor;
+    }
+    
+    public Node(int value){
+        this.value=value;
+        ancestor=null;
+    }
+
+    public int getValue() {
```
Comment (imorlowska) [inline krzeelzb/assignment-2/src/main/java/Node.java]: nit: from a design point of view - can the value change? Or should it stay the same once the node is created? If there's no need for the consumer of this class to modify this value then there is no need to expose the public setter. :)

File: krzeelzb/assignment-2/src/test/java/TestClass.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+public class TestClass {
+    final BinaryTree tree=BinaryTreeRepository.getBinaryTree();
+    final int valueFive=5;
+    final int valueOne=1;
+    final int valueTen=10;
+
+    @Test
+    public void findNodeFiveReturnsTrue(){
+        Assert.assertNotNull(FindNode.findNode(tree,valueFive));
```
Comment (imorlowska) [inline krzeelzb/assignment-2/src/test/java/TestClass.java]: How about also checking that the returned node is correct (ie. the .value and .ancestor fields are set correctly, or comparing the object)?

File: krzeelzb/assignment-2/src/test/java/TestClass.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    @Test
+    public void findNodeTenReturnsFalse(){
```
Comment (imorlowska) [inline krzeelzb/assignment-2/src/test/java/TestClass.java]: Maybe in the future use more descriptive names (or comments). "findNodeTenReturnsFalse" doesn't tell us 'why' and actually it looks for 'null'. :) Maybe something like "checkFindNodeReturnsNullForMissingKey" or similar.

File: krzeelzb/assignment-2/src/main/java/CommonAncestor.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import java.util.List;
+
+public class CommonAncestor {
+
+    public static Node getCommonAncestor(BinaryTree tree, Node nodeOne, Node nodeTwo){
+        List<Node> ancestorsNodeOne=FindAncestors.findAncestors(tree,nodeOne);
+        List<Node> ancestorsNodeTwo=FindAncestors.findAncestors(tree,nodeTwo);
```
[the remark this answers is not in the document]
Comment (imorlowska) [reply krzeelzb/assignment-2/src/main/java/CommonAncestor.java]: This is the only comment that really needs attention. The remaining ones can be treated as warnings for future assignments.

**Comment by imorlowska** [comment]
A quick reminder - the deadline to merge this solution was yesterday. Please resolve the comments and merge the final solution as soon as possible.

File: krzeelzb/assignment-2/src/main/java/BinaryTreeRepository.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        Node node2=new Node(2);
+        Node node3=new Node(3);
+        Node node7=new Node(7);
+        Node node4=new Node(4);
+
+        tree.leaves.get(0).setAncestor(node2); // from node 1 to 2
+        tree.leaves.get(1).setAncestor(node2); // from node 6 to 2
+        tree.leaves.get(2).setAncestor(node3); // from node 5 to 3
+        tree.leaves.get(3).setAncestor(node4); // from node 8 to 4
+        tree.leaves.get(0).getAncestor().setAncestor(node3); // from node 2 to 3
```
[the remark this answers is not in the document]
Comment (krzeelzb) [reply krzeelzb/assignment-2/src/main/java/BinaryTreeRepository.java]: You're right! I am changing it

File: krzeelzb/assignment-2/src/main/java/FindAncestors.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class FindAncestors {
+    public static List findAncestors(BinaryTree tree, Node value){
```
[the remark this answers is not in the document]
Comment (krzeelzb) [reply krzeelzb/assignment-2/src/main/java/FindAncestors.java]: Changed

**Comment by krzeelzb** [comment]
Thanks for the review. I thought that the deadlines are for the pull requests not merging. Sorry for that.

**Comment by krzeelzb** [comment]
Also sorry for my late replays, but I have busy time at Uni, a lot of projects to accomplish and stuff:(

**Comment by imorlowska** [comment]
Please merge this into master

**Pull request merged**